Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs

```diff
@@ -171,5 +191,153 @@ private DomainModel.Models.TestConduct.TestAttendees TestAttendee()
+
+          //Configuring Application User
+            ApplicationUser user = new ApplicationUser() { Email = userName, UserName = userName };
+            await _userManager.CreateAsync(user);
+            var applicationUser = await _userManager.FindByEmailAsync(user.Email);
+
+          //Creating questions
+            var question1 = CreatequestionAC(true, "This is in Category 1", category1.Id, 0);
+            var question2 = CreatequestionAC(true, "This is in Category 2", category2.Id, 0);
+            await _questionRepository.AddSingleMultipleAnswerQuestionAsync(question1, applicationUser.Id);
+            await _questionRepository.AddSingleMultipleAnswerQuestionAsync(question2, applicationUser.Id);
+
+            var AllQuestions = await _questionRepository.GetAllQuestionsAsync(user.Id);
+            List<Question> ListAC = AllQuestions.ToList();
+    
+            var test = CreateTest("Maths");
+            await _testRepository.CreateTestAsync(test);
+
+            //Adding categories to test
+            testCategory.TestId = test.Id;
+            testCategory.CategoryId = category1.Id;
+            await _trappistDbContext.TestCategory.AddAsync(testCategory);
+            await _trappistDbContext.SaveChangesAsync();
+            testCategory = new TestCategory();
+            testCategory.TestId = test.Id;
+            testCategory.CategoryId = category2.Id;
+            await _trappistDbContext.TestCategory.AddAsync(testCategory);
+            await _trappistDbContext.SaveChangesAsync();
+
+            List<QuestionAC> questionListAC = new List<QuestionAC>();
+            var questionDetailList = Mapper.Map<List<Question>, List<QuestionDetailAC>>(ListAC);
+            foreach (var question in questionDetailList)
+            {
+                QuestionAC questionAC = new QuestionAC();
+                question.IsSelect = true;
+                questionAC.Question = question;
+                questionListAC.Add(questionAC);
+            }
+     
+            await  _testRepository.AddTestQuestionsAsync(questionListAC, test.Id);
+            var questionACList = await _testRepository.GetAllTestCategoryQuestionsByIdAsync(test.Id,category1.Id);
+            Assert.Equal(1,questionACList.Count);
+            Assert.Equal(2, _trappistDbContext.TestQuestion.Count());
+            Assert.True(questionACList[0].Question.IsSelect == true);
+        }
+
+        private DomainModel.Models.Category.Category CreateCategory()
+        {
+            var category = new DomainModel.Models.Category.Category
+            {
+                CategoryName = "Test Category"
+            };
+            return category;
+        }
+        /// <summary>
+        ///Test case for  getting test details containing categories only
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task GetTestDetails()
+        {
+            TestAC testAC = new TestAC();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable 'testAC'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1854)